### ISSUE TITLE
HTML parser: output adds extra spans

### ISSUE THREAD

**Issue by bentranmyriota** [remarkjs/react-markdown#460]
Hi guys,
I found an issue with HTML parser parsing inline HTML in the markdown file. I used the default configuration as mentioned in the [Parsing HTML ](https://github.com/rexxars/react-markdown#parsing-html) section.

**Markdown content:**
```
Something is not right!
<span>Are you sure?</span>
<span>Yes</span>
```
**Rendered output in the browser:**
```
<p>Something is not right!
<span></span>Are you sure?
<span></span><span></span>Yes
</p>
```

Any help is appreciated!
Thanks

**Comment by bentranmyriota** [comment]
I did more investigation. The output from `remark-parse` after the source is parsed, looks like below for the inline html:
```
type: "paragraph",
children: [
   0.{type: "html", value: "<span>", position: Position}
   1: {type: "text", value: "Are you sure?", position: Position}
   2: {type: "html", value: "</span>", position: Position}
   3: {type: "text", value: "↵", position: Position}
   4: {type: "html", value: "<span>", position: Position}
   5: {type: "text", value: "Yes", position: Position}
   6: {type: "html", value: "</span>", position: Position}
]

```

Any idea of how to resolve this?

**Comment by ChristianMurphy** [comment]
this will be resolved by https://github.com/rexxars/react-markdown/pull/428

**Comment by wooorm** [comment]
This may also be fixed on 5.0.0, [with this change](https://github.com/remarkjs/react-markdown/commit/22a5e49ae9dde84257d043f033d1b640b78c7b34#diff-e48a0b02aa7a8f65a453b849e7345933ee0072fb6d0c2a109c83f43cc8fae8c6). Could someone confirm whether that works? Otherwise it’s just the HTML parser being funky and the plan is for that to be replaced next major.

**Comment by bentranmyriota** [comment]
I can confirm that the issue still persists.
### Input
```
Something is not right!
<span>Are you sure?</span>
<span>Yes</span>
```
### Output
```
<p>Something is not right!
<span></span>Are you sure?
<span></span><span></span>Yes</p>
```

**Comment by danestves** [comment]
I can confirm that i'm with this issue too:

### Input
```html
<span role="img" aria-label="party popper">🎉</span>
```

### Output
```html
<span>
  <span role="img" aria-label="party popper"></span>
  🎉
  <span></span>
</span>
```

And we really need this to work because we are transforming the markdown to a11y format

**Comment by justnewbee** [comment]
right... i also ran into this issue, and we really need it to work

react-markdown version is `5.0.3`

### input

```html
<button data-xx>xx</button>
```

### output

```html
<span><button data-xx=""></button></span>xx<span></span>
```

**Comment by justnewbee** [comment]
I cannot wait... so I wrote a very small markdown component based on micromark - https://www.npmjs.com/package/@alicloud/rc-markdown
It ships gfm and directive but no other extensions for the moment, suites my need now.

**Comment by wooorm** [comment]
You could also look at https://github.com/remarkjs/react-remark.
Fwiw, if you’re having trouble with types, then PRs are appreciated.

**Comment by Gervwyk** [comment]
Same issue on our side:
`react-markdown` version is `5.0.3`

##### input

```html
Score: <span style="color: #237804;" ><b>100</b></span>
```

##### output

```html
Score: <span style="color: #237804;" ><b>>93.6</b></span>
```html
<span>Score: <span><span style="color: #237804;"></span></span><b>93.6</b></span>
```

Anything we can do to help fix this?

**Comment by wooorm** [comment]
@Gervwyk Thanks for offering to help. Specifically, this issue is solved by https://github.com/remarkjs/react-markdown/issues/549. Alternatively, there are many ways to contribute, through code, financially, etc: https://github.com/remarkjs/.github/blob/main/contributing.md#contributions

**Comment by wooorm** [comment]
This should be solved by landing GH-563 today, which will soon be released in v6.0.0!
